### ISSUE TITLE
Occasional error from QAbstractItemModel events upon test tear down

### ISSUE THREAD

**Issue by kitchoi** [enthought/traitsui#986]
The following error occurs once from this travis build (https://travis-ci.org/github/enthought/traitsui/jobs/706873557) :
```
======================================================================
FAIL: test_run (test_all_examples.TestExample) (file_path='/home/travis/build/enthought/traitsui/examples/demo/Advanced/ListStrEditor_demo.py')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/enthought/traitsui/integrationtests/test_all_examples.py", line 293, in test_run
    run_file(file_path)
AttributeError: 'NoneType' object has no attribute 'auto_add'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/travis/build/enthought/traitsui/integrationtests/test_all_examples.py", line 300, in test_run
    file_path, exc, message
AssertionError: Executing /home/travis/build/enthought/traitsui/examples/demo/Advanced/ListStrEditor_demo.py failed with exception 'NoneType' object has no attribute 'auto_add'
 Traceback (most recent call last):
  File "/home/travis/build/enthought/traitsui/integrationtests/test_all_examples.py", line 293, in test_run
    run_file(file_path)
  File "/home/travis/build/enthought/traitsui/integrationtests/test_all_examples.py", line 277, in run_file
    exec(content, globals)
  File "<string>", line 50, in <module>
  File "/home/travis/build/enthought/traitsui/integrationtests/test_all_examples.py", line 238, in replaced_configure_traits
    process_cascade_events()
  File "/home/travis/.edm/envs/traitsui-test-3.6-pyqt/lib/python3.6/contextlib.py", line 88, in __exit__
    next(self.gen)
  File "/home/travis/.edm/envs/traitsui-test-3.6-pyqt/lib/python3.6/site-packages/traitsui/tests/_tools.py", line 67, in store_exceptions_on_all_threads
    raise exceptions[0]
  File "/home/travis/.edm/envs/traitsui-test-3.6-pyqt/lib/python3.6/site-packages/traitsui/qt4/list_str_model.py", line 53, in rowCount
    if editor.factory.auto_add:
AttributeError: 'NoneType' object has no attribute 'auto_add'

```

This is similar to the ones seen in https://github.com/enthought/traitsui/issues/854 for `ImageEnumEditor` and the one worked around in https://github.com/enthought/traitsui/pull/897 for `TabularEditor`.
The fix for #854 was to replace/unset the instance of `QAbstractItemModel` from the view widget in the editor's dispose. See https://github.com/enthought/traitsui/pull/974

**Comment by kitchoi** [comment]
Related to #907 as well.

**Comment by kitchoi** [comment]
This error no longer appears. Closing.

More info:
Part of that was this workaround being put in place which avoided the issue: https://github.com/enthought/traitsui/blob/f6887000284fe8195002faa9d06f0ad0e2f21a7f/integrationtests/test_all_examples.py#L233-L237
The workaround has now been removed and the AttributeError should have been solved by #1286.
